docs/PLUGIN_AUTHORING.md

```diff
@@ -17,152 +17,144 @@ See the License for the specific language governing permissions and
-GetMetricTypes(ConfigType) ([]MetricType, error)
-```
-The plugin uses the default values given in the ConfigPolicy so a config file doesn't need to be passed in for these rules. An example use case would be for the URL the Apache Collector collects from. Disclaimer: Two namespaces can't have rules with the same key name. E.g. you can't have the key "username" for /intel/foo/bar and a different "username" for /intel/foo/mock. They would need unique keys.
+For example `ACME` org writing a `water` plugin, gather `usage` in gallons from multiple locations.
 
-### Writing a processor plugin
-A Snap processor plugin allows filtering, aggregation, transformation, etc of collected telemetry data. To complaint with processor plugin interfaces defined in Snap, a processor plugin must implement the following methods:
-```
-GetConfigPolicy() (*cpolicy.ConfigPolicy, error)
-Process(contentType string, content []byte, config map[string]ctypes.ConfigValue) (string, []byte, error)
 ```
-### Writing a publisher plugin
-A Snap publisher plugin allows publishing processed telemetry data into a variety of systems, databases, and monitors through Snap metrics. To compliant with metric types and plugin interfaces defined in Snap, a publisher plugin must implement the following methods:
+/acme/water/usage
+/acme/water/NewYork/usage
+/acme/water/NewYork/rainfall
+/acme/water/NewYork/Bronx/usage
+/acme/water/NewYork/Chelsea/usage
+/acme/water/Portland/usage
+/acme/water/Portland/rainfall
+/acme/water/Portland/Hollywood/usage
+/acme/water/Portland/Pearl/usage
 ```
-GetConfigPolicy() (*cpolicy.ConfigPolicy, error)
-Publish(contentType string, content []byte, config map[string]ctypes.ConfigValue) error
-```
-### Exposing a plugin
-Creating the main program to serve the newly written plugin as an external process in main.go. By defining "Plugin.PluginMeta" with plugin specific settings, the newly created plugin may have its setting to override Snap global settings. Please refer to [a sample](https://github.com/intelsdi-x/snap/blob/master/plugin/collector/snap-plugin-collector-mock1/main.go) to see how main.go is written. You may browse [snap global settings](https://github.com/intelsdi-x/snap/blob/master/snapd.go#L45-L119).
 
-Building main.go generates a binary executable. You may choose to sign the executable with our [plugin signing](https://github.com/intelsdi-x/snap/blob/master/docs/PLUGIN_SIGNING.md).
+A plugin can have any number of internal plugin namespace. It's up to the plugin author on how to organize the metrics in a meaningful way. This information should be documented in the README for ease of usage.
 
-### Localization
-All comments and READMEs within the plugin code should be in English.  For different languages, include appropriate translation files within the plugin package for internationalization.
+The metric namespace cannot contain [Unicode character U+1F422](http://www.fileformat.info/info/unicode/char/1f422/index.htm) i.e. :turtle:
```

Comment:
I would remove this line.
1. Actually it can contain it. 
2. This is "kinda" an easter egg.